### PR TITLE
test: improve SILGen coverage on Windows

### PR DESCRIPTION
**Pull request by compnerd**
Make these SILGen tests pass on windows too

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @gottesmm

**Comment by compnerd** [comment]
CC: @slavapestov @jrose-apple - at the suggestion of @gottesmm

File: test/SILGen/witness_tables.swift [hunk 1]
```diff
@@ -1,9 +1,9 @@
 
-// RUN: %target-swift-emit-silgen -module-name witness_tables  -I %S/Inputs -enable-source-import %s -disable-objc-attr-requires-foundation-module -enable-objc-interop > %t.sil
+// RUN: %target-swift-emit-silgen -emit-sorted-sil -module-name witness_tables -I %S/Inputs -enable-source-import %s -disable-objc-attr-requires-foundation-module -enable-objc-interop > %t.sil
```
Comment (jrose-apple) [inline test/SILGen/witness_tables.swift]: Why the change to sorting here?

**Review by jrose-apple — COMMENTED** [review]
Mostly seems reasonable to me.

File: test/SILGen/witness_tables.swift (hunk 1 above)
Comment (compnerd) [reply test/SILGen/witness_tables.swift]: The order differs between Linux/Darwin and Windows.  I don't think that there is a way to get both sides to work without the sorting.  Perhaps there is some other approach that I didn't consider.

Comment (jrose-apple) [reply test/SILGen/witness_tables.swift]: Hm. I'm pretty sure the order is supposed to be based on source, so that's a little suspicious.

Comment (gottesmm) [reply test/SILGen/witness_tables.swift]: My fear about this is that the need for it could show a source of non-determinism.

Comment (compnerd) [reply test/SILGen/witness_tables.swift]: @jrose-apple, empirical observation indicates that this is not the case.  There is certainly a possibility of there being a bug in SILGen.

**Comment by gottesmm** [comment]
I want this to land. @compnerd can you split out the -emit-sorted-sil changes. I do think it is important to figure out /why/ this is happening. I talked with Jordan and we think the rest of it is ok.

**Comment by compnerd** [comment]
Splintered the unsorted bits into #21493

**Comment by compnerd** [comment]
Going to close this, however, this is still unfixed.  The change in #22308 fixes the sorting issue.
